fem/gridfunc.cpp

```diff
@@ -462,15 +462,26 @@ const
    fes->GetElementDofs(i, dofs);
    fes->DofsToVDofs(vdim-1, dofs);
    const FiniteElement *FElem = fes->GetFE(i);
-   MFEM_ASSERT(FElem->GetMapType() == FiniteElement::VALUE,
-               "invalid FE map type");
    int dof = FElem->GetDof();
    Vector DofVal(dof), loc_data(dof);
    GetSubVector(dofs, loc_data);
-   for (int k = 0; k < n; k++)
+   if (FElem->GetMapType() == FiniteElement::VALUE)
```

Comment:
Does this address https://github.com/mfem/mfem/issues/1037?

Reply:
Partially.  This PR extended `GridFunction::GetValue`.  I added a commit today which addresses the limitation in `CurlInterpolator`.

Issue #1037 also mentions a problem in the `VectorFECurlIntegrator` (and possibly others).  I'll try to address these in a separate PR.